File: FileSystemManipulation/FileSystemManipulationUITests/FileSystemManipulationUITests.m
#import <XCTest/XCTest.h>
#import <FileSystemManipulator/FileSystemManipulator.h>

@interface FileSystemManipulationUITests : XCTestCase
@property (nonatomic, strong) XCUIApplication *app;
@end

@implementation FileSystemManipulationUITests

- (void)setUp {
    [super setUp];
    self.continueAfterFailure = NO;

    FileSystemManipulatorConfiguration *configuration = [[FileSystemManipulatorConfiguration alloc] init];

    NSURL *dataURL = [self writeStringToTemporaryFile:@"Injected Data!"];
    [configuration copyFileAtURL:dataURL toRelativePath:@"Documents/data.txt"];

    self.app = [[XCUIApplication alloc] init];
    self.app.launchEnvironment = [configuration generateLaunchEnvironment];
    [self.app launch];
}

- (void)testShowsTextLoadedFromFileInDocumentsDirectory {
    XCTAssert(self.app.staticTexts[@"Injected Data!"].exists);
}

#pragma mark - Helpers

- (NSURL *)writeStringToTemporaryFile:(NSString *)string {
    NSURL *fileURL = [NSURL fileURLWithPath:[NSTemporaryDirectory() stringByAppendingPathComponent:[[NSProcessInfo processInfo] globallyUniqueString]]];
    [string writeToURL:fileURL atomically:NO encoding:NSUTF8StringEncoding error:NULL];
    return fileURL;
}

@end

File: Swift/FileSystemManipulation/FileSystemManipulationUITests/FileSystemManipulationUITests.swift
import XCTest
import FileSystemManipulator

class FileSystemManipulationUITests: XCTestCase {
    let app = XCUIApplication()

    override func setUp() {
        super.setUp()
        continueAfterFailure = false

        let configuration = FileSystemManipulatorConfiguration()

        do {
            let dataURL = try writeStringToTemporaryFile("Injected Data!")
            configuration.copyFileAtURL(dataURL, toRelativePath: "Documents/data.txt")
        } catch let error {
            XCTFail("Failed to write data file: \(error)")
        }

        app.launchEnvironment = configuration.generateLaunchEnvironment()
        app.launch()
    }

    func testShowsTextLoadedFromFileInDocumentsDirectory() {
        XCTAssert(app.staticTexts["Injected Data!"].exists)
    }

    private func writeStringToTemporaryFile(string: String) throws -> NSURL {
        let filePath = (NSTemporaryDirectory() as NSString).stringByAppendingPathComponent(NSProcessInfo.processInfo().globallyUniqueString)
        let fileURL = NSURL(fileURLWithPath: filePath)
        try string.writeToURL(fileURL, atomically: false, encoding: NSUTF8StringEncoding)
        return fileURL
    }
}
